examples/manipulation/behavior_cloning.py

```
@@ -0,0 +1,414 @@
+            self._collect_with_rl_teacher()
+            end_time = time.time()
+            forward_time = end_time - start_time
+
+            # Training steps for both action and pose prediction
+            total_action_loss = 0.0
+            total_pose_loss = 0.0
+            num_batches = 0
+
+            start_time = time.time()
+            generator = self._buffer.get_batches(self._cfg.get("num_mini_batches", 4), self._cfg["num_epochs"])
+            for batch in generator:
+                # Forward pass for both action and pose prediction
+                pred_action = self._policy(batch["rgb_obs"], batch["robot_pose"])
+                pred_left_pose, pred_right_pose = self._policy.predict_pose(batch["rgb_obs"])
+
+                # Compute action prediction loss
+                action_loss = F.mse_loss(pred_action, batch["actions"])
+
+                # Compute pose estimation loss (position + orientation)
+                pose_left_loss = self._compute_pose_loss(pred_left_pose, batch["object_poses"])
+                pose_right_loss = self._compute_pose_loss(pred_right_pose, batch["object_poses"])
+                pose_loss = pose_left_loss + pose_right_loss
+
+                # Combined loss with weights
+                total_loss = action_loss + pose_loss
+
+                # Backward pass
+                self._optimizer.zero_grad()
+                total_loss.backward()
+                self._optimizer.step()
+                torch.nn.utils.clip_grad_norm_(self._policy.parameters(), self._cfg["max_grad_norm"])
+
+                total_action_loss += action_loss
+                total_pose_loss += pose_loss
+                num_batches += 1
+
+            end_time = time.time()
+            backward_time = end_time - start_time
+
+            # Compute average losses
+            avg_action_loss = total_action_loss / num_batches if num_batches > 0 else 0.0
+            avg_pose_loss = total_pose_loss / num_batches if num_batches > 0 else 0.0
```

It looks like you should rather raise an exception (at init) or consider that the average action / pose loss is undefined (`float("nan")`) if the number of batches is zero?